### ISSUE TITLE
Documentation: {{#each - should reference how to access the index and/or key?

### ISSUE THREAD

**Issue by TotallyInformation** [ractivejs/ractive#3162]
## Description:
The documentation for `{{#each ` doesn't indicate how to access the current index and/or the current key. This is confusing for beginners.

## Versions affected:
0.9.10

## Platforms affected:
All

## Reproduction:
https://ractive.js.org/api/#each-sections

I know that I can access the current key by using `@key`, not sure about the index. I expect there are other ways too.

Example code:

```
        <h2>IoT Devices - Last Update</h2>

        <table>
            <thead>
                <tr>
                    <th>Device ID</th><th>Last Update</th><th>Count</th><th>Location</th><th>Battery</th><th>Signal</th><th>Temperature</th><th>Humidity</th><th>Command</th>
                </tr>            
            </thead>
            <tbody>
                {{#each homie}}
                <tr>
                    <td data-label="Device ID">{{@key}}</td>
                    <td data-label="Last Update" style="{{timeAgo(updated)}}">{{updated}}<span title="Online?">{{online}}</span></td>
                    <td data-label="Count">{{count}}</td>
                    <td data-label="Location">{{location}}</td>
                    <td data-label="Battery">{{battery}}</td>
                    <td data-label="Signal">{{signal}}</td>
                    <td data-label="Temperature">{{temperature}}</td>
                    <td data-label="Humidity">{{humidity}}</td>
                    <td data-label="Command">{{command}}<span title="Command Unit">{{cmdUnit}}</span></td>
                </tr>
                {{/each}}
            </tbody>
        </table>
```

**Comment by ghost** [comment]
The index is accessed by `@index`. It's documented in detail under the [special references](https://ractive.js.org/api/#special-references) section.

Regular [section](https://ractive.js.org/api/#sections) blocks have index/key access documented inline since accessing them is part of its syntax. However, for special references, it made sense grouping them together in the docs. Before, they were inlined in things they're normally used. But this made it hard to know what special references are out there without reading through their accompanying feature.

**Comment by krityan** [comment]
The way to think about it is, will the user know to go from `#each` to special references and will they draw a connection between each and index/key. If no then `#each` should mention them.

The user is going to search for `#each` when looking for details about it.

Sometimes repeating small bit of information for a better user experience is worthwhile, or at least including a link to other pieces of relevant information. Currently reading the docs provides a pretty fragmented and incomplete feeling overview of Ractive which makes it harder for people in our team to pick it up.

**Comment by PaulMaly** [comment]
@krityan Thanks, for your response. Yes, right now docs are not so cool as it should be. I think even old docs were much more informative. I believe we will improve the docs with your help too. Sometimes I use old docs to refresh some knowledge: [0.8 docs](https://ractive.js.org/v0.x/0.8/get-started)
There are many things which weren't changed in newer versions but described much better in old docs.

**Comment by ghost** [comment]
Speaking of docs, feel free to send a PR over at https://github.com/ractivejs/ractivejs.github.io 😄

**Comment by TotallyInformation** [comment]
I'm still learning - a lot! - so hopefully it is useful to get a fresh pair of eyes on the docs. Not yet ready to mess up your docs with a PR but if things go according to plan, I may be able to in a few weeks.

There is a lot to learn and quite a few different ways to achieve the same thing it would seem - nice and flexible but slightly painful for learning.

Also, it is hard to know whether I am following best practice or just bringing outdated ideas from working on other frameworks. Even so, I've already been able to easily reproduce something I did in MoonJS and start work on a table display component & I love the flexibility. I'm expanding on my uibuilder for Node-RED and so I really need the component features to be able to create easy to consume ui components that can be driven from Node-RED (with the data coming in via Socket.IO). And I really like the flexibility of being able to create components on the server back end or the front end as needed.
